docs/integrations/ml-dist.rst

```diff
@@ -0,0 +1,512 @@
+
+We will begin with loading in our data to a pandas dataframe, so we can have
+a look at it and make adjustments accordingly.
+
+.. code-block:: python
+
+    data = pd.read_csv("/path/to/sensor.csv")
+
+    # Shows us the first five rows of the dataset
+    data.head(5)
+
+                     timestamp  sensor_00  sensor_01  sensor_02  sensor_03  sensor_04  sensor_05  sensor_06  sensor_07  ...  sensor_43  sensor_44  sensor_45  sensor_46  sensor_47  sensor_48  sensor_49  sensor_50  sensor_51  machine_status
+    0   0  2018-04-01 00:00:00   2.465394   47.09201    53.2118  46.310760   634.3750   76.45975   13.41146   16.13136  ...   41.92708  39.641200   65.68287   50.92593  38.194440   157.9861   67.70834   243.0556   201.3889          NORMAL
+    1   1  2018-04-01 00:01:00   2.465394   47.09201    53.2118  46.310760   634.3750   76.45975   13.41146   16.13136  ...   41.92708  39.641200   65.68287   50.92593  38.194440   157.9861   67.70834   243.0556   201.3889          NORMAL
+    2   2  2018-04-01 00:02:00   2.444734   47.35243    53.2118  46.397570   638.8889   73.54598   13.32465   16.03733  ...   41.66666  39.351852   65.39352   51.21528  38.194443   155.9606   67.12963   241.3194   203.7037          NORMAL
+    3   3  2018-04-01 00:03:00   2.460474   47.09201    53.1684  46.397568   628.1250   76.98898   13.31742   16.24711  ...   40.88541  39.062500   64.81481   51.21528  38.194440   155.9606   66.84028   240.4514   203.1250          NORMAL
+    4   4  2018-04-01 00:04:00   2.445718   47.13541    53.2118  46.397568   636.4583   76.58897   13.35359   16.21094  ...   41.40625  38.773150   65.10416   51.79398  38.773150   158.2755   66.55093   242.1875   201.3889          NORMAL
+
+    [5 rows x 55 columns]
+
+    # We can see that the first column has no name
+    data.rename(columns={data.columns[0]: "id"}, inplace=True)
+
+    # Now let's see if the dataset has any missing values
+    data.isna().sum().sum()
+
+    [1] 344574
+
+    # Ok so we have 344574 missing values, let's fill those up
+    missing_values = -1.0
+    data = data.fillna(missing_values)
+
+Now that we have filled the missing values of the dataset, we can insert the
+data to our CrateDB instance.
+
+.. code-block:: python
+
+    data.to_csv("sensor_new.csv", index=False)
+
+    conn_str = "host=localhost user=crate port=5432"
+
+    # This should only be run once, as it will duplicate the data in the table otherwise.
+    with connect(conn_str) as conn:
+        with conn.cursor() as cur:
+            query = "COPY ml.pump_sensors FROM 'file:///path/to/sensor_new.csv'"
+            cur.execute(query)
+
+Training A Model
+----------------
+
+First, we will load the data into our python context with ``Psycopg2`` and
+``Pandas``
+
+.. code-block:: python
+
+    import pandas.io.aql as sqlio
+    from psycopg2 import connect
+
+
+    with connect(conn_str) as conn:
+        query = "SELECT * FROM ml.pump_sensors;"
+        df = sqlio.read_sql_query(query, conn)
```

Comment:
In this code block, and following code blocks, could you add a brief inline comment above each block of code about what is happening/what you're doing? Here it is simple enough, but later when you start doing stuff with sklearn/tf it can be a little difficult to follow. something like:

```suggestion
.. code-block:: python

    import pandas.io.aql as sqlio
    from psycopg2 import connect

    
    # Connecting to CrateDB and retrieving the pump sensor data.
    with connect(conn_str) as conn:
        query = "SELECT * FROM ml.pump_sensors;"
        df = sqlio.read_sql_query(query, conn)
```